larray/core/array.py

```diff
@@ -547,6 +547,92 @@ def _doc_agg_method(func, by=False, long_name='', action_verb='perform', extra_a
 
 obj_isnan = np.vectorize(lambda x: x != x, otypes=[bool])
 
+
+def equal(a1, a2, rtol=0, atol=0, nan_equals=False):
+    """
+    Compares two arrays element-wise and returns array of booleans.
+
+    Parameters
+    ----------
+    a1, a2 : LArray-like
+        Input arrays. aslarray() is used on non-LArray inputs.
+    rtol : float or int, optional
+        The relative tolerance parameter (see Notes). Defaults to 0.
+    atol : float or int, optional
+        The absolute tolerance parameter (see Notes). Defaults to 0.
+    nan_equals: boolean, optional
+        Whether or not to consider nan values at the same positions in the two arrays as equal.
+        By default, an array containing nan values is never equal to another array, even if that other array
+        also contains nan values at the same positions. The reason is that a nan value is different from
+        *anything*, including itself. Defaults to False.
+
+    Returns
+    -------
+    LArray
+        Returns True if the arrays are equal (even in the presence of NaN).
```

Comment:
either remove "(even in the presence of NaN)" or replace it with something vaguely like "(the definition of which depending on the given arguments)". I don't really like this particular wording, but I suppose you get what I mean.